repo.go

```diff
@@ -803,6 +798,12 @@ func (r *Repo) Snapshot() error {
 	return r.SnapshotWithExpires(data.DefaultExpires("snapshot"))
 }
 
+func (r *Repo) snapshotManifests() []string {
+	// Note: root pinning is not supported in Spec 1.0.19.
```

Comment:
Should we parameterize/flag protect this? Using go-tuf as a library causes dependent libraries to print out the root pinning warning. It would be nice to let users decide whether to include root
https://github.com/theupdateframework/go-tuf/issues/203

Reply:
Can this by default remove root.json? If we log warn about it, we should default to avoid the log.

But still think perhaps we should add a parameter to include root when calling `Snapshot`

Reply:
What would be the motivation for a user wanting the `root.json` included? IMO it doesn't matter much how it's configured as long as the default is to not include it. But if there's not a clear motivation for providing the option I'd say keep it simple and don't add the option.

Reply:
I removed root.json from the snapshot. I had to change a number of tests to make them pass and also regenerate test data.

As of https://github.com/theupdateframework/go-tuf/pull/143, verification of root.json in snapshot has been removed from the client (replaced with that log line). I'm not sure I see the value in keeping a config option around to write root.json to snapshot if it's not actually verified in the client. The only thing that comes to mind is whether old clients will break against new repos written without root.json in snapshots, but changing the default with configuration via a build flag or by adding an options API is also a breaking change on repo side. I would lean toward making the repo-side breaking change for simplicity and to match the 1.0 spec sooner. What do you think @asraa?

Reply:
>  I would lean toward making the repo-side breaking change for simplicity and to match the 1.0 spec sooner. 

This is reasonable to me!